website/docs/en/guide/basic/static-assets.mdx

```diff
@@ -37,17 +39,32 @@ Import with [alias](/guide/advanced/alias) is also available:
 ```tsx
 import logo from '@/static/logo.png';
 
+console.log(logo); // "/static/logo.[hash].png"
+
 export default = () => <img src={logo} />;
 ```
 
-You can also combine [URL](https://developer.mozilla.org/docs/Web/API/URL) with [import.meta.url](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/import.meta) which are native ESM features to import static assets.
+### URL assets
+
+Rsbuild supports using JavaScript's native [URL](https://developer.mozilla.org/docs/Web/API/URL) and [import.meta.url](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/import.meta) to import static assets.
 
 ```tsx
 const logo = new URL('./static/logo.png', import.meta.url).href;
 
+console.log(logo); // "/static/logo.[hash].png"
+
 export default = () => <img src={logo} />;
 ```
 
+If you use `new URL()` to reference `.js` or `.ts` files, they will be treated as URL assets and will not be processed by Rsbuild's built-in `swc-loader`.
+
```

Comment:
Shall we need to add a guide on how to treat it as a URL asset and have it processed by the built-in `swc-loader` in Rsbuild?

Reply:
I have no idea how to support it, do you have any suggestions?

Reply:
no idea +1 😢

Reply:
We can find a solution and include a guide in another PR.